src/main/java/fi/digitraffic/tis/vaco/rules/RuleResultsListener.java

```diff
@@ -115,26 +123,34 @@ protected CompletableFuture<Boolean> handleDeadLetter(JsonNode jsonNode) {
         return CompletableFuture.supplyAsync(() -> {
             logger.info("Dead letter Queue Message {}", jsonNode);
 
+            Optional<Finding> finding = createDeadLetterQueueFinding(jsonNode);
+            finding.ifPresent(findingRepository::create);
+
             Optional<String> taskPublicId = Optional.ofNullable(jsonNode)
                 .filter(node -> node.has("task"))
                 .map(node -> node.get("task").get("publicId"))
                 .map(JsonNode::asText);
 
+            Optional<String> entryPublicId = getEntryPublicId(jsonNode);
+
+            Optional<Entry> entry = entryPublicId
+                .flatMap(entryService::findEntry);
+
             return taskPublicId
                 .flatMap(taskService::findTask)
                 .map(task -> {
                     taskService.markStatus(task, Status.FAILED);
+                    entry.ifPresent(e ->
+                        taskService.trackTask(e, task, ProcessingState.COMPLETE));
                     return true;
                 }).orElse(false);
         }).whenComplete((deadLetterProcessingSuccess, maybeEx) -> {
             if (maybeEx != null) {
-                logger.warn("Handling deal letter queue message failed due to unhandled exception", maybeEx);
+                logger.warn("Handling dead letter queue message failed due to unhandled exception", maybeEx);
```

Comment:
There should probably be `return;` after this, as `deadLetterProcessingSuccess` is always false at this point anyway...hmm, not sure actually, please check if you can.